### ISSUE TITLE
Add logic for purple-3

### ISSUE THREAD

**Issue by SolarAquarion** [hoehermann/purple-gowhatsapp#54]
It currently fails to build because it looks for purple, and not purple-3. Perhaps add some logic to the MAKEFILE to say PURPLE=purple-3

**Comment by hoehermann** [comment]
I want to support purple 3 one day. I mainly write this plug-in so I can use it myself. And I don't use Pidgin 3, yet. Contributions are welcome, of course. :)

**Comment by SolarAquarion** [comment]
I have an initial patch on my system, but, i don't know how to go in the logic of pidgin 3 because some of the headers have been removed

**Comment by hoehermann** [comment]
Yes, that describes the challenge quite well. Preprocessor switches are necessary to discriminate the version numbers. You can look at these projects of EionRobb's for examples on how to add support for both libpurple versions.

* https://github.com/EionRobb/purple-discord/
* https://github.com/EionRobb/skype4pidgin/

**Comment by hoehermann** [comment]
I looked into this the other day. I actually managed to build https://github.com/hoehermann/purple-signald/tree/purple-3 against purple-3. It has a number of issues, but nothing too severe. However, the purple-3 developers said that the purple-3 interfaces are not mature, yet: https://discord.com/channels/351558764417122305/624712166968131595/1013234593240973352 Relying on them would be unwise. I put this on hold until purple-3 is ready.

**Comment by hoehermann** [comment]
**Update 2026:** Interfaces in purple3 have indeed changed and significantly so. It is necessary to re-write the glue code of this plug-in from scratch. I published some preliminary work [here](https://codeberg.org/hoehermann/purple3-whatsmeow).
